### PR TITLE
Change return type for "abs_diff"

### PR DESCRIPTION
**Pull request by gmlueck**
Change `abs_diff` so that its return type is the same as the type of its parameters.  This makes the signature consistent with `abs`.

Closes internal issue 655.

### CODE REVIEW

**Comment by gmlueck** [comment]
IMPORTANT!

This is currently a "stacked" PR on top of #440.  Once #440 is merged, this PR should be retargeted to the "SYCL-2020/master" branch.

**Review by keryell — APPROVED** [review]
Thanks.

**Review by TApplencourt — APPROVED** [review]

**Review by AerialMantis — APPROVED** [review]

**Comment by gmlueck** [comment]
Hi @keryell,

I think this PR should not have been merged yet because it is "stacked" on top of #440.  Merging it just merges it to my private branch.  We should wait for #440 to be merged, then I'll retarget these PRs to the "SYCL-2020/master" branch.  Once that happens, we can merge them.

I converted the remaining "stacked" PRs to draft status for now to prevent them from being merged.

**Comment by keryell** [comment]
Oops. I merged this by error. :-(
@gmlueck can you remove this commit from https://github.com/KhronosGroup/SYCL-Docs/tree/gmlueck/gentype-funcs-v2 and reopen the PR, probably requiring some dummy commit on the source branch?
Sorry for the mess...

**Comment by gmlueck** [comment]
Yes, I will fix it up tomorrow.

**Comment by gmlueck** [comment]
I couldn't find a way to reopen a PR that is already merged, so I created a new PR instead: #458.  Could those people who approved this PR also approve #458?

I removed the incorrectly merged commits from https://github.com/KhronosGroup/SYCL-Docs/tree/gmlueck/gentype-funcs-v2.

**Pull request merged**
